src/dotnet/src/Mil.Navy.Nrl.Norm/NormFile.cs

```diff
@@ -26,13 +29,24 @@ public string Name
         {
             get
             {
-                var buffer = new char[FILENAME_MAX];
-                if (!NormFileGetName(_handle, buffer, FILENAME_MAX))
+                var buffer = new byte[FILENAME_MAX];
+                var bufferHandle = GCHandle.Alloc(buffer, GCHandleType.Pinned);
+
+                try
+                {
+                    var bufferPtr = bufferHandle.AddrOfPinnedObject();
+                    if (!NormFileGetName(_handle, bufferPtr, FILENAME_MAX))
+                    {
+                        throw new IOException("Failed to get file name");
+                    }
+                } 
+                finally
                 {
-                    throw new IOException("Failed to get file name");
+                    bufferHandle.Free();
                 }
+                
                 buffer = buffer.Where(c => c != 0).ToArray();
-                return new string(buffer);
+                return new string(buffer.Select(Convert.ToChar).ToArray());
```

Comment:
@lbargaoanu Thanks for the feedback. I will update `NormFile.Name`.

Reply:
As the examples shows, the `GCHandle` is not needed. And using a pointer saves you an allocation because you can use `stackalloc`. But in either case, you can simply pass the managed object, PInvoke does the pinning.

Reply:
@lbargaoanu Pushed update to `NormFile.Name` that uses `stackalloc` and `PtrToStringAnsi`. Please verify if the comments have been addressed and if there are other recommendations.